samples/DXFeedOptionChain/Program.cs

```diff
@@ -0,0 +1,107 @@
+/// </summary>
+[SuppressMessage("ReSharper", "MethodSupportsCancellation")]
+internal abstract class Program
+{
+    public static async Task Main(string[] args)
+    {
+        if (args.Length != 5)
+        {
+            Console.WriteLine("usage: DXFeedOptionChain <address> <ipf-file> <symbol> <nStrikes> <nMonths>");
+            Console.WriteLine("       <ipf-file> is endpoint address");
+            Console.WriteLine("       <ipf-file> is name of instrument profiles file");
+            Console.WriteLine("       <symbol>   is the product or underlying symbol");
+            Console.WriteLine("       <nStrikes> number of strikes to print for each series");
+            Console.WriteLine("       <nMonths>  number of months to print");
+            return;
+        }
+
+        var argAddress = args[0];
+        var argIpfFile = args[1];
+        var argSymbol = args[2];
+        var nStrikes = int.Parse(args[3], CultureInfo.InvariantCulture);
+        var nMonths = int.Parse(args[4], CultureInfo.InvariantCulture);
+
+        var feed = DXEndpoint.Create().Connect(argAddress).GetFeed();
+
+        // Subscribe to trade to learn instrument last price.
+        Console.WriteLine($"Waiting for price of {argSymbol} ...");
+        using var cts = new CancellationTokenSource();
+        cts.CancelAfter(TimeSpan.FromSeconds(1));
+        var trade = await feed.GetLastEventAsync<Trade>(argSymbol, cts.Token);
+
+        var price = trade.Price;
+        Console.WriteLine($"Price of {argSymbol} is {price.ToString(CultureInfo.InvariantCulture)}");
+
+        Console.WriteLine($"Reading instruments from {argIpfFile} ...");
+        var instruments = new InstrumentProfileReader().ReadFromFile(argIpfFile).ToList();
+
+        Console.WriteLine("Building option chains ...");
+        var chains = OptionChainsBuilder<InstrumentProfile>.Build(instruments).Chains;
+        if (!chains.TryGetValue(argSymbol, out var chain))
+        {
+            Console.WriteLine($"No chain found for symbol {argSymbol}");
+            return;
+        }
+
+        nMonths = Math.Min(nMonths, chain.GetSeries().Count);
+        var seriesList = chain.GetSeries().Take(nMonths).ToList();
+
+        Console.WriteLine("Requesting option quotes ...");
+        var quotes = new Dictionary<InstrumentProfile, Task<Quote>>();
+        foreach (var series in seriesList)
+        {
+            var strikes = series.GetNStrikesAround(nStrikes, price);
+            foreach (var strike in strikes)
+            {
+                if (series.Calls.TryGetValue(strike, out var call))
+                {
+                    quotes[call] = feed.GetLastEventAsync<Quote>(call.Symbol);
+                }
+
+                if (series.Puts.TryGetValue(strike, out var put))
+                {
+                    quotes[put] = feed.GetLastEventAsync<Quote>(put.Symbol);
+                }
+            }
+        }
+
+        await Task.WhenAll(quotes.Values);
```

Comment:
Originaly it was:
// ignore timeout and continue to print retrieved quotes even on timeout
Promises.allOf(quotes.values()).awaitWithoutException(1, TimeUnit.SECONDS);

Reply:
done